clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -10211,8 +10211,8 @@ Value *CodeGenFunction::EmitSVETupleSetOrGet(const SVETypeFlags &TypeFlags,
          "Expects TypleFlag isTupleSet or TypeFlags.isTupleSet()");
 
   unsigned I = cast<ConstantInt>(Ops[1])->getSExtValue();
-  auto *SingleVecTy = dyn_cast<llvm::ScalableVectorType>(
-                      TypeFlags.isTupleSet() ? Ops[2]->getType() : Ty);
+  auto *SingleVecTy = cast<llvm::ScalableVectorType>(
```

Comment:
I am not sure we want the compile to crash  that way, at least that is my impression.
Should this be an if  statement
if(auto *SingleVecTy = cast<llvm::ScalableVectorType>....){
value *Idx = ConstantInt::get(CGM.Int64Ty,
                                I * SingleVecTy->getMinNumElements());
}
Is it possible to add a test for this change too?

Reply:
Thank you @CarolineConcatto for reviews. I have updated patch. Could you please review again? 

>>Is it possible to add a test for this change too?

The issues are reported by Static Analyzer tool, I am not sure how we can add a test here to verify the fix.

@tahonermann, any opinion?